### ISSUE TITLE
Pyright hanging with unannotated script

### ISSUE THREAD

**Issue by ekolve** [microsoft/pyright#7791]
Running the following snippet will cause pyright to hang and never return from the CLI. My system configuration is:

MacOS 14.4.1 M1, Python 3.12, Pyright 1.1.360, Node v22.0.0, default pyright configuration (no pyrightconfig.json).

```bash
pyright test.py 
```

```python
def func2(a, name):
    return a

def func1(h, ids):
    for _ in ids:
        h = func2(h, 1)
        h = func2(h, 2)
        h = func2(h, 3)
        h = func2(h, 4)
        h = func2(h, 5)
        h = func2(h, 6)
        h = func2(h, 7)
        h = func2(h, 8)
        h = func2(h, 9)
        h = func2(h, 10)
        h = func2(h, 11)
        h = func2(h, 12)
        h = func2(h, 13)
        h = func2(h, 14)
        h = func2(h, 15)
        h = func2(h, 16)
        h = func2(h, 17)
        h = func2(h, 18)
        h = func2(h, 19)
        h = func2(h, 20)
        h = func2(h, 21)
        h = func2(h, 22)
        h = func2(h, 23)
        h = func2(h, 24)
        h = func2(h, 25)
        h = func2(h, 26)
        h = func2(h, 27)
        h = func2(h, 28)
        h = func2(h, 29)
        h = func2(h, 30)
        h = func2(h, 31)
        h = func2(h, 32)
        h = func2(h, 33)
        h = func2(h, 34)
        h = func2(h, 35)
        h = func2(h, 36)
        h = func2(h, 37)
        h = func2(h, 38)
        h = func2(h, 39)
```

If the script is limited to just the following - pyright will return in ~0.4 seconds:

```python
def func2(a, name):
    return a

def func1(h, ids):
    for _ in ids:
        h = func2(h, 1)
        h = func2(h, 2)
        h = func2(h, 3)
        h = func2(h, 4)
        h = func2(h, 5)
        h = func2(h, 6)
        h = func2(h, 7)
        h = func2(h, 8)
        h = func2(h, 9)
        h = func2(h, 10)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.361, which I just published.
